### ISSUE TITLE
Add a brief tutorial on transfer learning

### ISSUE THREAD

**Issue by yashjain-99** [rocketmlhq/rmldnn#4]
I'm trying to use rmldnn to classify images from a Kaggle dataset of birds, but I'm having trouble doing so. We normally utilise transfer learning for datasets this huge, and in Tensorflow it's very simple: we just create a base model with weights, and then onto the last layer, we add layers as necessary to get the desired amount of classes as output.
But when I added a pre-trained model (say VGG16 from the resources) and adjusted the input and output layers to match the dataset, it didn't provide any results or return any errors.
So, if a tutorial on transfer learning could be added, that would be extremely beneficial.
Kaggle dataset on which I was working: https://www.kaggle.com/datasets/gpiosenka/100-bird-species

**Comment by ssbotelh** [comment]
Hi @yashjain-99, yes, we should be adding a full tutorial on transfer learning for image classification soon.
But just for clarification, the json files available under resources contain network descriptions only, not model parameters. For transfer learning, you'd also need the corresponding h5 file from Keras/Tensorflow, which contains the actual model parameters. We will provide that soon as well.

However, I'm curious by what you meant by "it didn't provide any results or return any errors". Were you able to run rmldnn? If so, could you show your config file and what you got as output? Thanks

**Comment by yashjain-99** [comment]
About issue when using same network as given in documentation of MNIST classification but with modification(Attached NN).
rmldnn docker image version: rmldnn latest
Configuration file: config_training.json and config_test.json(attached)
input data file:  https://www.kaggle.com/datasets/gpiosenka/100-bird-species
Command run to reproduce the error:  sudo docker run -u $(id -u):$(id -g) -v ${PWD}:/home/ubuntu -w /home/ubuntu --rm \
    rocketml/rmldnn:latest rmldnn --config=config_training.json
Expected behavior: It should have ran and should have given low accuracy as model is not complex enough for this large data.
Screenshots: Attached
Desktop: Ubuntu(22.04) , Docker(20.10.12)
All files available in zip below. Kindly refer that and for dataset use above mentioned kaggle link.
[Files.zip](https://github.com/rocketmlhq/rmldnn/files/8710650/Files.zip)
If you require any additional information or files, please let me know.

**Comment by adavanisanti** [comment]
@yashjain-99  could you try the following configuration file? Please note these changes:
1. `resize` in `data` --> `transforms` to match the input layer in the network 
2. `NLL` loss function because this is a classification problem. `Dice` loss in your current configuration works for image segmentation problem.

```
{
    "neural_network": {
        "outfile": "out.txt",
        "num_epochs": 40,
        "layers": "./net.json",
        "checkpoints": {
            "save": "./model/",
            "interval": 20
        },
        "data": {
            "input_type": "images",
            "target_type": "labels",
            "input_path":      "./train",
            "test_input_path": "./valid",
            "batch_size": 8,
            "grayscale": false,
            "preload": false,
            "transforms": [
                { "resize": [224, 224] }
            ]
        },
        "optimizer": {
            "type": "Adam",
            "learning_rate": 1e-4
        },
        "loss": {
            "function": "NLL"
        }
    }
}
```

**Comment by yashjain-99** [comment]
Yes, it worked. I will provide the accuracy when the training is finished because it is taking far too long, even though I reduced the number of labels from 400 to 100.
Thanks for help. @ssbotelh @adavanisanti 
screenshot attached:
![Screenshot from 2022-05-18 01-38-57](https://user-images.githubusercontent.com/91678870/168965417-a81c5d87-7534-4dbe-8389-8216e98edf13.png)

**Comment by yashjain-99** [comment]
An error came up during evaluation: Target -1 out of bound.(Screenshots attached)
![Screenshot from 2022-05-18 01-53-54](https://user-images.githubusercontent.com/91678870/168967316-acf9ec2b-b4d2-4ac1-a727-746b01f582ac.png)
![Screenshot from 2022-05-18 01-54-48](https://user-images.githubusercontent.com/91678870/168967330-8855d27b-adc0-45e8-8d3b-20bea3c569f7.png)
![Screenshot from 2022-05-18 02-33-06](https://user-images.githubusercontent.com/91678870/168973167-a9351955-6b00-4f32-a4df-f644ff679e78.png)

**Comment by ssbotelh** [comment]
Hi @yashjain-99, this error is caused by a mistake in the dataset. Notice how the directory
``BLACK & YELLOW  BROADBILL`` has an extra space in the train set, but not in the test or validation sets. As a result, test samples from that category will not have a matching label, thus the invalid ``-1`` target label. 

Please rename this directory in the training set by removing the extra space, and retry your run. In future releases, I'll make sure this error happens early and with a clear error message.

**Comment by yashjain-99** [comment]
Yes thanks a lot @ssbotelh , It is working all fine now.
